builtin/common/misc_helpers.lua

```diff
@@ -198,9 +198,44 @@ function table.indexof(list, val)
 	return -1
 end
 
+table.indexof = table.index_of
 assert(table.indexof({"foo", "bar"}, "foo") == 1)
 assert(table.indexof({"foo", "bar"}, "baz") == -1)
 
+--------------------------------------------------------------------------------
+function table.count(t)
+	local i = 0
+	for _ in pairs(t) do
+		i = i + 1
+	end
+	return i
+end
+
+--------------------------------------------------------------------------------
+function table.random(t)
+	return t[math.random(#t)]
```

Comment:
This doesn't work with tables that are not lists.
This restriction is not documented either.

Reply:
fixed